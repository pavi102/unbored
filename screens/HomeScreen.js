import React from 'react'
import { View, Text, StyleSheet, TouchableOpacity, StatusBar } from 'react-native'


const HomeScreen = ({ navigation }) => {
    return (
        <View style={styles.container}>
            <StatusBar style="auto" />
            <View style={styles.textContainer}>
                <Text style={styles.mainText}>Unbored</Text>
                <View style={styles.horizontalLine} />
                <TouchableOpacity onPress={() => navigation.navigate('Activity')}>
                    <Text style={styles.mainText}>What to do</Text>
                    <Text style={styles.mainText}>right now?</Text>
                </TouchableOpacity>
                <View style={styles.horizontalLine} />
                <TouchableOpacity onPress={() => navigation.navigate('TellMeMore')}>
                    <Text style={styles.mainText}>Something</Text>
                    <Text style={styles.mainText}>more</Text>
                </TouchableOpacity>
                <View style={styles.horizontalLine} />
            </View>
        </View>
    );
}


const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#34393b',
        alignItems: 'center',
    },
    textContainer: {
        backgroundColor: '#34393b',
        margin: '15%',
        alignItems: 'center',

    },
    mainText: {
        fontSize: 30,
        fontFamily: 'monospace',
        color: '#a3a6ad',
        alignSelf: 'center',
        paddingHorizontal: '10%'

    },
    horizontalLine: {
        borderBottomColor: '#a3a6ad',
        borderBottomWidth: 2,
        alignSelf: 'stretch',
        marginHorizontal: "12%",
        marginTop: "10%",
        marginBottom: "20%",
    },
    item: {
        padding: 10,
        fontSize: 18,
        height: 44,
    },
});

export default HomeScreen;