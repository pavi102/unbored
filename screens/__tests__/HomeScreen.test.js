import React from 'react';
import {cleanup, render} from '@testing-library/react-native'
import HomeScreen from '../HomeScreen'

describe('Home screen', () => {
    afterEach(cleanup)
    it('should render correctly', () => {
        const toJson = render(<HomeScreen />)
        expect(toJson).toMatchSnapshot();
    })
})
    