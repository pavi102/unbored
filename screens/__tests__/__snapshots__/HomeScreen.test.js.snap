// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#34393b",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#34393b",
        "margin": "15%",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#a3a6ad",
          "fontFamily": "monospace",
          "fontSize": 30,
          "paddingHorizontal": "10%",
        }
      }
    >
      Unbored
    </Text>
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "borderBottomColor": "#a3a6ad",
          "borderBottomWidth": 2,
          "marginBottom": "20%",
          "marginHorizontal": "12%",
          "marginTop": "10%",
        }
      }
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#a3a6ad",
            "fontFamily": "monospace",
            "fontSize": 30,
            "paddingHorizontal": "10%",
          }
        }
      >
        What to do
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#a3a6ad",
            "fontFamily": "monospace",
            "fontSize": 30,
            "paddingHorizontal": "10%",
          }
        }
      >
        right now?
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "borderBottomColor": "#a3a6ad",
          "borderBottomWidth": 2,
          "marginBottom": "20%",
          "marginHorizontal": "12%",
          "marginTop": "10%",
        }
      }
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#a3a6ad",
            "fontFamily": "monospace",
            "fontSize": 30,
            "paddingHorizontal": "10%",
          }
        }
      >
        Something
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#a3a6ad",
            "fontFamily": "monospace",
            "fontSize": 30,
            "paddingHorizontal": "10%",
          }
        }
      >
        more
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "borderBottomColor": "#a3a6ad",
          "borderBottomWidth": 2,
          "marginBottom": "20%",
          "marginHorizontal": "12%",
          "marginTop": "10%",
        }
      }
    />
  </View>
</View>
`;
